joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

head_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints: 
      - head_j0
      - head_j1
    gains:
      head_j0: {p: 100,  d: 1, i: 1, i_clamp: 1} 
      head_j1: {p: 100,  d: 1, i: 1, i_clamp: 1}

zlift_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints: 
      - zlift_j0
    pid_gains:
      zlift_j0: {p: 100.0, i: 0.01, d: 10.0}
    gains:
      zlift_j0: {p: 100,  d: 1, i: 1, i_clamp: 1} 

torso_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints: 
      - torso_j0
      - torso_j1
    pid_gains:
      torso_j0: {p: 100.0, i: 0.01, d: 10.0}
      torso_j1: {p: 100.0, i: 0.01, d: 10.0}
    gains:
      torso_j0: {p: 100,  d: 1, i: 1, i_clamp: 1}
      torso_j1: {p: 100,  d: 1, i: 1, i_clamp: 1}  

right_arm_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints: 
      - right_arm_j0
      - right_arm_j1
      - right_arm_j2
      - right_arm_j3
      - right_arm_j4
      - right_arm_j5
      - right_arm_j6
    gains:
      right_arm_j0: {p: 100,  d: 1, i: 1, i_clamp: 1}
      right_arm_j1: {p: 100,  d: 1, i: 1, i_clamp: 1} 
      right_arm_j2: {p: 100,  d: 1, i: 1, i_clamp: 1}
      right_arm_j3: {p: 100,  d: 1, i: 1, i_clamp: 1}  
      right_arm_j4: {p: 100,  d: 1, i: 1, i_clamp: 1}
      right_arm_j5: {p: 100,  d: 1, i: 1, i_clamp: 1}
      right_arm_j6: {p: 100,  d: 1, i: 1, i_clamp: 1}   

right_hand_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - right_hand_j0
      - right_hand_j1
      - right_hand_j2
      - right_hand_j3
      - right_hand_j4
    gains:
      right_hand_j0: {p: 100,  d: 1, i: 1, i_clamp: 1}
      right_hand_j1: {p: 100,  d: 1, i: 1, i_clamp: 1} 
      right_hand_j2: {p: 100,  d: 1, i: 1, i_clamp: 1}
      right_hand_j3: {p: 100,  d: 1, i: 1, i_clamp: 1}  
      right_hand_j4: {p: 100,  d: 1, i: 1, i_clamp: 1}  
    
left_arm_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints: 
      - left_arm_j0
      - left_arm_j1
      - left_arm_j2
      - left_arm_j3
      - left_arm_j4
      - left_arm_j5
      - left_arm_j6
    gains:
      left_arm_j0: {p: 100,  d: 1, i: 1, i_clamp: 1}
      left_arm_j1: {p: 100,  d: 1, i: 1, i_clamp: 1} 
      left_arm_j2: {p: 100,  d: 1, i: 1, i_clamp: 1}
      left_arm_j3: {p: 100,  d: 1, i: 1, i_clamp: 1}  
      left_arm_j4: {p: 100,  d: 1, i: 1, i_clamp: 1}
      left_arm_j5: {p: 100,  d: 1, i: 1, i_clamp: 1}
      left_arm_j6: {p: 100,  d: 1, i: 1, i_clamp: 1}   

left_hand_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - left_hand_j0
      - left_hand_j1
      - left_hand_j2
      - left_hand_j3
      - left_hand_j4
    gains:
      left_hand_j0: {p: 100,  d: 1, i: 1, i_clamp: 1}
      left_hand_j1: {p: 100,  d: 1, i: 1, i_clamp: 1} 
      left_hand_j2: {p: 100,  d: 1, i: 1, i_clamp: 1}
      left_hand_j3: {p: 100,  d: 1, i: 1, i_clamp: 1}  
      left_hand_j4: {p: 100,  d: 1, i: 1, i_clamp: 1}  
      
#gazebo_ros_control:
#  pid_gains:
#    zlift_j0: {p: 100.0, i: 0.01, d: 10.0}
#    torso_j0: {p: 100.0, i: 0.01, d: 10.0}
#    torso_j1: {p: 100.0, i: 0.01, d: 10.0}
#    torso_j2: {p: 100.0, i: 0.01, d: 10.0}
